modules/postgres/14.18.bcr.4/overlay/BUILD.bazel

```diff
@@ -0,0 +1,1311 @@
+    "src/port/win32ntdll.c",
+    "src/port/win32pread.c",
+    "src/port/win32pwrite.c",
+    "src/port/win32security.c",
+    "src/port/win32setlocale.c",
+    "src/port/win32stat.c",
+    "src/port/open.c",
+    "src/port/dirmod.c",
+]
+
+LIBCOMMON_BASE = [
+    "src/common/base64.c",
+    "src/common/encnames.c",
+    "src/common/fe_memutils.c",
+    "src/common/file_perm.c",
+    "src/common/ip.c",
+    "src/common/jsonapi.c",
+    "src/common/kwlookup.c",
+    "src/common/link-canary.c",
+    "src/common/logging.c",
+    "src/common/md5_common.c",
+    "src/common/md5.c",
+    "src/common/pg_get_line.c",
+    "src/common/pg_lzcompress.c",
+    "src/common/pgfnames.c",
+    "src/common/psprintf.c",
+    "src/common/restricted_token.c",
+    "src/common/rmtree.c",
+    "src/common/saslprep.c",
+    "src/common/scram-common.c",
+    "src/common/sha1.c",
+    "src/common/sha2.c",
+    "src/common/sprompt.c",
+    "src/common/string.c",
+    "src/common/stringinfo.c",
+    "src/common/unicode_norm.c",
+    "src/common/username.c",
+    "src/common/wait_error.c",
+    "src/common/wchar.c",
+]
+
+LIBCOMMON_NOT_WINDOWS = [
+    "src/common/cryptohash.c",
+    "src/common/hmac.c",
+]
+
+LIBCOMMON_WINDOWS = [
+    "src/common/cryptohash.c",
+    "src/common/hmac.c",
+]
+
+LIBPQ_BASE = [
+    "src/interfaces/libpq/fe-auth-scram.c",
+    "src/interfaces/libpq/fe-auth.c",
+    "src/interfaces/libpq/fe-connect.c",
+    "src/interfaces/libpq/fe-exec.c",
+    "src/interfaces/libpq/fe-lobj.c",
+    "src/interfaces/libpq/fe-misc.c",
+    "src/interfaces/libpq/fe-print.c",
+    "src/interfaces/libpq/fe-protocol3.c",
+    "src/interfaces/libpq/fe-secure.c",
+    "src/interfaces/libpq/fe-secure-common.c",
+    "src/interfaces/libpq/fe-trace.c",
+    "src/interfaces/libpq/legacy-pqsignal.c",
+    "src/interfaces/libpq/libpq-events.c",
+    "src/interfaces/libpq/pqexpbuffer.c",
+]
+
+LIBPQ_NOT_WINDOWS = [
+    "src/interfaces/libpq/fe-secure-openssl.c",
+]
+
+LIBPQ_WINDOWS = [
+    "src/interfaces/libpq/pthread-win32.c",
+    "src/interfaces/libpq/win32.c",
+]
+
+cc_library(
+    name = "pq",
+    hdrs = LIBPQ_PUBLIC_HDRS,
+    includes = ["src/include", "src/interfaces"],
+    srcs = LIBPQ_BASE + select({
+        "@platforms//os:linux": LIBPQ_NOT_WINDOWS + LIBCOMMON_BASE + LIBCOMMON_NOT_WINDOWS + LIBPORTS_BASE + LIBPORTS_LINUX,
+        "@platforms//os:macos": LIBPQ_NOT_WINDOWS + LIBCOMMON_BASE + LIBCOMMON_NOT_WINDOWS + LIBPORTS_BASE + LIBPORTS_MACOS,
+        "@platforms//os:windows": LIBPQ_WINDOWS + LIBCOMMON_BASE + LIBCOMMON_WINDOWS + LIBPORTS_BASE + LIBPORTS_WINDOWS,
+    }),
+    implementation_deps = [
+        ":include",
+    ],
+    # copts = [
+    #     "-include",
+    #     "$(location :src/include/pg_config.h)",
+    #     "-include",
+    #     "$(location :src/include/pg_config_os.h)",
+    # ],
+    deps = select({
+        "@platforms//os:windows": [],
+	":use_boringssl_setting": ["@boringssl//:ssl", "@boringssl//:crypto"],
```

Comment:
This is not correct. You'll likely wanna make a wrapper `cc_library` target for these deps which uses this select statement to switch between boringssl and openssl and then replace them in the `//conditions:default` section.
E.g.
```starlark
cc_library(
    name = "pq_ssl_crypto",
    deps = select({
        ":use_boringssl_setting": ["@boringssl//:ssl", "@boringssl//:crypto"],
        "//conditions:default": ["@openssl//:ssl", "@openssl//:crypto"],
    }),
    tags = ["manual"],
)
```

Also, if you can run buildifier on the file that'd be sweet!